generate/codegen/modulewriter.go

```diff
@@ -85,6 +91,171 @@ func (m *ModuleWriter) WriteTypeAliases() {
 		}
 		cw.WriteLineF("type %s = %s", fa.GName, fa.Type.GoName(&m.Module, false))
 	}
+
+	for _, fa := range m.StructAliases {
+		if fa.DocURL != "" {
+			cw.WriteLine(fmt.Sprintf("// %s [Full Topic]", fa.Description))
+			cw.WriteLine(fmt.Sprintf("//\n// [Full Topic]: %s", fa.DocURL))
+		}
+		cw.WriteLineF("type %s = unsafe.Pointer", fa.GName)
+	}
+}
+
+func (m *ModuleWriter) WriteStructs() {
+	if len(m.Structs) == 0 {
+		return
+	}
+
+	filePath := filepath.Join(m.PlatformDir, m.Module.Package, "structs.gen.go")
+	os.MkdirAll(filepath.Dir(filePath), 0755)
+	f, err := os.Create(filePath)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	cw := &CodeWriter{Writer: f, IndentStr: "\t"}
+	cw.WriteLine(AutoGeneratedMark)
+	cw.WriteLine("package " + m.Module.Package)
+
+	cw.WriteLine("import (")
+
+	// imports
+	var imports = set.New("unsafe")
+	for _, fa := range m.FuncAliases {
+		imports.AddSet(fa.GoImports())
+	}
+	cw.Indent()
+	imports.ForEach(func(value string) {
+		if value != "github.com/progrium/macdriver/macos/objc" {
+			cw.WriteLine("\"" + value + "\"")
+		}
+	})
+	cw.UnIndent()
+	cw.WriteLine(")")
+
+	for _, s := range m.Structs {
+		// if Ref type, allias to unsafe.Pointer
+		if strings.HasSuffix(s.Name, "Ref") {
+			if s.DocURL != "" {
+				cw.WriteLine(fmt.Sprintf("// %s [Full Topic]", s.Description))
+				cw.WriteLine(fmt.Sprintf("//\n// [Full Topic]: %s", s.DocURL))
+			}
+
+			cw.WriteLineF("type %s unsafe.Pointer", s.GoName)
+			continue
+		}
+		log.Println("skipping struct", s.Name)
+	}
+}
+
+func shouldSkipFunction(f *Function) bool {
+	if f.Deprecated {
+		return true
+	}
+	if hasBlockParam(f.Parameters) {
+		return true
+	}
+	if _, ok := map[string]bool{
+		"CGDirectDisplayCopyCurrentMetalDevice":            true,
```

Comment:
can we have reasons for why these are skipped in comments?

Reply:
good call, largely an artifact of development and these having edge cases, I'll pull this out for now as I'm preferring to flag things off but I agree we should include comments with any exceptions.